File: js/sumPairs.js
exports.sumPairs = function(integerArrays, desiredSum) {
    let output = []
    for (let i=1; i < integerArrays.length; i++) { 
        for (let int of integerArrays.slice(i)) { //find all possible pairings
            if (integerArrays[i] + int === desiredSum) {
                output.push([integerArrays[i], int])
            }
        }}
    
    if (output.length != 0) {
        return output} 
    
        
    return 'unable to find pair'

};
//console.log(exports.sumPairs([1,2,3,4,5], 9))
//console.log(Array(2))
//console.log(exports.sumPairs([1,1,3], 2)))

File: python/sum_pairs.py
def sum_pairs(list_integers, desired_sum):
    """
    Given a list of integers and a desired sum, return a list
    of list of pairs of integers that sum up to the desired sum. 
    If no pairs satisfied the desired sum, return 'unable to find pairs'.
    """
    output = []
    for i in range(len(list_integers)):
        for elt in list_integers[i:]: #paired with all elts after each elt, this get all the possible pairs
            if list_integers[i] + elt == desired_sum:
                output.append([list_integers[i],elt])

    return output if output else 'unable to find pair' #empty list has a bool of False